### PR TITLE
Linked list library

### PR DESCRIPTION
**Pull request by Melestyl**
(almost) All the linked list functions are now separate from server, to be used in client as well

### CODE REVIEW

File: Common/linked_list.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#ifndef PAS_LINKED_LIST_H
+#define PAS_LINKED_LIST_H
+
+#include "common.h"
+#define bool char
+
```
Comment (3Dai1) [inline Common/linked_list.h:6]:
```suggestion
#define true 1
#define false 0
```

File: Common/linked_list.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#ifndef PAS_LINKED_LIST_H
+#define PAS_LINKED_LIST_H
+
+#include "common.h"
```
Comment (3Dai1) [inline Common/linked_list.h:4]:
```suggestion
```

File: Common/linked_list.h (hunk 1 above)
Comment (3Dai1) [reply Common/linked_list.h:6]: You have to remove this 2 line from Common.h

**Review by 3Dai1 — APPROVED** [review]

File: Common/linked_list.h (hunk 1 above)
Comment (Melestyl) [reply Common/linked_list.h:6]: flemme

File: Common/linked_list.h (hunk 2 above)
Comment (Melestyl) [reply Common/linked_list.h:4]: flemme

**Pull request merged**